ouroboros-consensus-byron/src/Ouroboros/Consensus/Byron/Ledger/Forge.hs

```diff
@@ -67,7 +67,7 @@ forgeEBB
 forgeEBB cfg curSlot curNo prevHash =
         mkByronBlock (byronEpochSlots byronConfig)
       . CC.Block.ABOBBoundary
-      . CC.reAnnotateBoundary (byronProtocolMagicId byronConfig)
+      . CC.reAnnotateABOBBoundary (byronProtocolMagicId byronConfig)
```

Comment:
Can you show what difference this makes? I'm not sure I think it does. Previously, we were reannotating the EBB (`ebb :: ABoundaryBlock`): so we serialise the EBB and use that bytestring as the annotation. Now, you're reannotating the EBB as part of the `ABOBBoundary` sum (`ABOBBoundary (ebb :: ABoundaryBlock)`), so you serialise the `ABlockOrBoundary`, which means you'll have an extra `encodeListLen 2 <> encodeWord 0` at the start of the bytestring, but then when we annotate during decoding, we slice those two extra bytes back off the annotation. The end result is that you encoded and dropped two extra bytes, without changing the end result. Or am I missing something?

Reply:
In practical terms, this removes the EBB hash disagreement between all three Chairman's cluster participant types:  legacy, proxy, rewrite.

In terms of code, the fix repairs just the `forgeEBB` code path, which is serialisation only (although it does a roundtrip to check deserialisation, but that's just a local validation).

On the decoding path, we're using a function that properly dispatches on the ABOB tag:

1. `decodeByronBlock`: https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-consensus-byron/src/Ouroboros/Consensus/Byron/Ledger/Serialisation.hs#L120
2. `decodeUnsizedHeader`: https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-consensus-byron/src/Ouroboros/Consensus/Byron/Ledger/Serialisation.hs#L306